### PR TITLE
[MRG] Adding updated tutorial to mention BIDS vs mne coordinate systems.

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #715 

To work on after #704 is merged.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by hoechenberger** [comment]
Awesome, @adam2392! Ping me when you want some feedback

**Comment by adam2392** [comment]
https://4136-89170358-gh.circle-artifacts.com/0/dev/auto_examples/convert_ieeg_to_bids.html#sphx-glr-auto-examples-convert-ieeg-to-bids-py

how's this for first pass?

Once CI passes, then it's just however we want to describe these issues in this tutorial.

**Comment by adam2392** [comment]
@hoechenberger any issues w/ this?

**Comment by hoechenberger** [comment]
There's a rendering issue:

Screen Shot 2021-03-03 at 19 59 28

Also I think this important info is somehow getting lost, being buried down there at the very end of the tutorial. Can we maybe move it up somehow? I think it deserves its own Info Box or something…

**Comment by adam2392** [comment]
> There's a rendering issue:
> 
> Screen Shot 2021-03-03 at 19 59 28

Why are my sphinx/rst skills so subpar :(.

> And you're sometimes talking about "MNE", sometimes about "MNE-Python"; I think this can be made more consistent :)
> 
> Also I think this important info is somehow getting lost, being buried down there at the very end of the tutorial. Can we maybe move it up somehow? I think it deserves its own Info Box or something…
agreed.

**Comment by hoechenberger** [comment]
> Why are my sphinx/rst skills so subpar :(.

Because RST is a horrible standard and Sphinx syntax is just as horrible too 🤯 My personal opinion ;)

File: examples/convert_ieeg_to_bids.py [hunk 1]
```diff
@@ -124,12 +125,31 @@
 print("The channel coordinates before writing into BIDS: ")
 pprint([x for x in zip(ch_names, pos)])
 
+###############################################################################
+# BIDS vs MNE-Python Coordinate Systems
+# -------------------------------------
+#
+# BIDS has many acceptable coordinate systems for iEEG, which can be viewed in
+# the `specification. <https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html>`_  # noqa
```
Comment (sappelhoff) [inline examples/convert_ieeg_to_bids.py:133]:
```suggestion
# `appendix VIII`_ of the BIDS specification.
```

Add to bottom of file:

`.. _appendix VIII: https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html`

File: examples/convert_ieeg_to_bids.py [hunk 2]
```diff
@@ -124,12 +125,31 @@
...
+# However, MNE-BIDS depends on MNE-Python and MNE-Python does not support all
+# these coordinate systems (yet).
+#
+# MNE-Python has a few tutorials on this topic:
+#   - `background on FreeSurfer <https://mne.tools/dev/auto_tutorials/source-modeling/plot_background_freesurfer_mne>`_  # noqa
+#   - `MNE-Python coordinate frames <https://mne.tools/dev/auto_tutorials/source-modeling/plot_source_alignment.html>`_  # noqa
```
Comment (sappelhoff) [inline examples/convert_ieeg_to_bids.py:139]:
```suggestion
#   - `background on FreeSurfer`_
#   - `MNE-Python coordinate frames`_
```

Add to bottom of file:
```
.. _background on FreeSurfer: https://mne.tools/dev/auto_tutorials/source-modeling/plot_background_freesurfer_mne
.. _MNE-Python coordinate frames: https://mne.tools/dev/auto_tutorials/source-modeling/plot_source_alignment.html
```

**Review by sappelhoff — COMMENTED** [review]
@adam2392 I think this should work.

**Comment by sappelhoff** [comment]
Don't forget to add this to the bottom of the file after merging my two suggestions.

```
.. _appendix VIII: https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html
.. _background on FreeSurfer: https://mne.tools/dev/auto_tutorials/source-modeling/plot_background_freesurfer_mne
.. _MNE-Python coordinate frames: https://mne.tools/dev/auto_tutorials/source-modeling/plot_source_alignment.html
```

**Comment by sappelhoff** [comment]
sometimes it's better to commit, then to comment and ask somebody to commit something :-D

**Comment by adam2392** [comment]
> sometimes it's better to commit, then to comment and ask somebody to commit something :-D

Thanks!

File: doc/whats_new.rst [hunk 3]
```diff
@@ -39,6 +39,7 @@ Enhancements
 - Some datasets out in the real world have a non-standard ``stim_type`` instead of a ``trial_type`` column in ``*_events.tsv``. :func:`mne_bids.read_raw_bids` now makes use of this column, and emits a warning, encouraging users to rename it, by `Richard Höchenberger`_ (:gh:`680`)
 - When reading data where the same event name or trial type refers to different event or trigger values, we will now create a hierarchical event name in the form of ``trial_type/value``, e.g. ``stimulus/110``, by `Richard Höchenberger`_ (:gh:`688`)
 - When reading data via :func:`mne_bids.read_raw_bids`, the channel names specified in the BIDS ``*_channels.tsv`` and ``*_electrodes.tsv`` files now always take precedence (and do not need to match) the channel names stored in the raw files anymore, by `Adam Li`_ and `Richard Höchenberger`_ (:gh:`691`, :gh:`704`)
+- Improve the ``convert ieeg to BIDS`` tutorial to include a note on how BIDS and MNE-Python coordinate frames are handled, by `Adam Li`_ (:gh:`717`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:42]:
```suggestion
- Improve the ``Convert iEEG data to BIDS`` tutorial to include a note on how BIDS and MNE-Python coordinate frames are handled, by `Adam Li`_ (:gh:`717`)
```

**Review by sappelhoff — APPROVED** [review]

**Review by hoechenberger — APPROVED** [review]

**Comment by adam2392** [comment]
Is there a reason it's not showing up in our renderings: https://mne.tools/mne-bids/dev/auto_examples/convert_ieeg_to_bids.html#

File: examples/convert_ieeg_to_bids.py [hunk 4]
```diff
@@ -124,13 +127,33 @@
...
+#   - `MNE-Python coordinate frames`_
+#
+# Currently, MNE-Python supports the ``mni_tal`` coordinate frame, which
+# corresponds to the ``fsaverage`` BIDS coordinate system. All other coordinate
+# frames in MNE-Python if written with :func:`mne_bids.write_raw_bids` are
+# written with coordinate system ``'Other'``. Note, then we suggest using
+# :func:`mne_bids.update_sidecar_json` to update the sidecar
+# ``*_coordsystem.json`` file to add additional information.
+
 ###############################################################################
```
Comment (sappelhoff) [inline examples/convert_ieeg_to_bids.py:151]: @adam2392 yes ... my fault: this bar of ################### needs to be removed

Comment (sappelhoff) [reply examples/convert_ieeg_to_bids.py:151]: and then the upper block of comments needs to be connected with the "step 2: formatting ...." block via a  # char

**Comment by sappelhoff** [comment]
> Is there a reason it's not showing up in our renderings: https://mne.tools/mne-bids/dev/auto_examples/convert_ieeg_to_bids.html#

fixed in https://github.com/mne-tools/mne-bids/commit/df53001b5c265c26aabc006fa1ae59b6a45f5073

File: examples/convert_ieeg_to_bids.py (hunk 4 above)
Comment (sappelhoff) [reply examples/convert_ieeg_to_bids.py:151]: see: https://github.com/mne-tools/mne-bids/pull/717#issuecomment-799448682

**Pull request merged**